Correct branch (jdiazpena:main) for the possible merge with main gemini3D repo

jdiazpena opened the pull request.

mattzett commented:
I've started merging Jules' changes first since they are less extensive.  We are almost done and I'll tackle this one next.

mattzett commented:
I'm currently working on this here:  https://github.com/gemini3d/gemini3d/tree/develop_FBI

This does not pass the short CI, I think because there have been some minor changes in energy equations solutions.  I also have not yet incorporated the different energy solution styles (i.e. all diffusion//source terms handled in a single parabolic solver).

mattzett commented:
It's mostly passing the short CI now; there are very minor changes in some of the examples in Te mostly.  I'd like to track down why this happens before writing it off as unimportant...

mattzett commented:
I don't yet have a replacement for the code outputting the conductances; however, there is a fork we are working on that has a more general approach to outputting auxiliary variables that I will eventually also merge --> @jdiazpena we can sync your fork with that merged version and then add the conductivity output back in at that point.

mattzett commented:
Long CI results below.  The actual differences that trigger these are quite small.  

The following tests FAILED:
	  6 - compare:output:GDI_periodic_lowres (Failed)       compare hourly
	 12 - compare:output:arcs (Failed)                      compare hourly
	 36 - compare:output:mini2dns_glow (Failed)             compare hourly
	 42 - compare:output:mini3d_fang (Failed)               compare hourly
	 48 - compare:output:mini3d_glow (Failed)               compare hourly
	 66 - compare:output:cusp_softprecip2D_Neu (Failed)     compare daily


 90/100 Test  #42: compare:output:mini3d_fang ..............................***Failed    0.17 sec
OK: gemini3d.compare: grid: /Users/zettergm/Projects/gemci/build/mini3d_fang == /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang
check_simsize:  reference /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/inputs/simsize.h5
check_simsize:  new /Users/zettergm/Projects/gemci/build/mini3d_fang/inputs/simsize.h5
check_simsize:  OK: /Users/zettergm/Projects/gemci/build/mini3d_fang/inputs/simsize.h5  48 12 16
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18000.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18000.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18060.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18060.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18120.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18120.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18180.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18180.000000.h5
MISMATCH: 20130220_18240.000000.h5 Te
 max diff:   1.445E+01 max & min ref:   2.729E+03   1.904E+02 max & min new:   2.729E+03   1.904E+02
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18240.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18240.000000.h5
 gemini3d.compare: MISMATCHED data at        2013           2          20   18240.000000000000     
MISMATCH: 20130220_18300.000000.h5 Te
 max diff:   3.384E+01 max & min ref:   2.726E+03   1.904E+02 max & min new:   2.726E+03   1.904E+02
 MISMATCH: 20130220_18300.000000.h5 v1   14.5319824    
auto-closing /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang/20130220_18300.000000.h5
auto-closing /Users/zettergm/Projects/gemci/build/mini3d_fang/20130220_18300.000000.h5
 gemini3d.compare: MISMATCHED data at        2013           2          20   18300.000000000000     
ERROR STOP gemini3d.compare output: FAIL /Users/zettergm/Projects/gemci/build/mini3d_fang != /Users/zettergm/Projects/gemci/build/test_ref/mini3d_fang

mattzett commented:
@jdiazpena I'm seeing the above minor Te differences in our test problems even with all the FBI and new vibcool rates switched off.  Do you have a sense for what might be causing these?  I know we did a fair bit of limiting in different places so that could create some small changes potentially.

jdiazpena commented:
Yes, but my changes should only work for extreme temperatures. My code
would limit some electron-neutral maxwell collisions that would have an
effect on Te. Same with the thermal conductivity. Lines 221 and 229 of
collisions.f90 are the thermal conductivity changes, where I made a limit.
Should not trigger I believe since it would only appear at higher
temperatures.

*-----------------------------------------------------------------------------------------*
*Joaquín M. Díaz Peña*
* PhD in Electrical Engineering*
Postdoctoral Researcher in Systems and Instrumentation for Space
Environments
Space and Planetary Exploration Laboratory
Universidad de Chile







On Thu, Apr 24, 2025 at 9:30 AM Matthew D. Zettergren <
***@***.***> wrote:

> *mattzett* left a comment (gemini3d/gemini3d#119)
> <https://github.com/gemini3d/gemini3d/pull/119#issuecomment-2827641300>
>
> @jdiazpena <https://github.com/jdiazpena> I'm seeing the above minor Te
> differences in our test problems even with all the FBI and new vibcool
> rates switched off. Do you have a sense for what might be causing these? I
> know we did a fair bit of limiting in different places so that could create
> some small changes potentially.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gemini3d/gemini3d/pull/119#issuecomment-2827641300>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXRWCZG32EDCBNS3ZIQ4LL23DRNXAVCNFSM6AAAAAB3XMRNRGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMRXGY2DCMZQGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mattzett commented:
@jdiazpena I did a quick test and by reverting the limiting I can get rid of these minor Te differences that were triggering CI failures.  What puzzles me is that I don't think any of these limits should be triggering for the test problems...

jdiazpena commented:
Yes, I remember this changes. I added the extra neutral equation (that is why it now includes Hydrogen in the calculation nn(:,:,:,4)*5.47e-15_wp*(1-1.35e-4*Tstmp2))), as well as capped the Nitrogen part since it would go negative. That is why we have differences at lower temperatures, I added an extra neutral to the equation.

jdiazpena commented:
I have 2D examples on my big desktop at home, they were even published so they would be perfect to test all these runs.

mattzett commented:
> I have 2D examples on my big desktop at home, they were even published so they would be perfect to test all these runs.

Sounds good could you put those in a cloud folder somewhere; I may need to alter them to be smaller in terms of number of cells so they run a bit faster.

mattzett commented:
@jdiazpena I think I got everything -- there are a few caveats including how the conductivity output data are handled, certain defaults that I've chosen, and the ```arcs``` CI test which is still failing due to temperature-limiting.
